templates/widget/__tests__/widget.test.ts

```diff
@@ -0,0 +1,33 @@
+import 'reflect-metadata';
+import { MessageClient, MessageService } from '@theia/core';
+import { ContainerModule, Container } from 'inversify';
+import { MockMessageClient, MockMessageService } from './mock-objects/mock-message-service';
+import { <%= params.extensionPrefix %>Widget } from '../<%= params.extensionPath %>-widget';
+import { render } from '@testing-library/react'
+
+describe('widget extension unit tests', () => {
+
+    let widget: any;
+
+    beforeEach(async () => {
+        const module = new ContainerModule( bind => {
+            bind(MessageClient).to(MockMessageClient).inSingletonScope();
+            bind(MessageService).to(MockMessageService).inSingletonScope();
```

Comment:
We can mock what we need:

```suggestion
            bind(MessageService).toConstantValue({
                info(message: string): void {
                    console.log(message);
                }
            } as MessageService);
```